libs/client-sdk/include/launchdarkly/client_side/data_source_status.hpp

```diff
@@ -5,202 +5,75 @@
 #include <functional>
 #include <memory>
 #include <optional>
-#include <ostream>
-#include <string>
+#include <utility>
 
 #include <launchdarkly/connection.hpp>
+#include <launchdarkly/data_sources/data_source_status_base.hpp>
 
 namespace launchdarkly::client_side::data_sources {
 
-class DataSourceStatus {
-   public:
-    using DateTime = std::chrono::time_point<std::chrono::system_clock>;
-
+/**
```

Comment:
Calling out as a theoretical breaking change.

 In previous Client SDK, you could call `IDataSourceStatusProvider::Status()` and get back a `class client_side::data_sources::DataSourceStatus`.

Now, you get back a `class common::data_sources::DataSourceStatusBase<DataSourceState>`. 

These classes should have the same API, but the concrete type has changed. 

If the caller did:
```c++
auto status = sdk.DataSourceStatus().Status();
if (auto err = status.LastError()) {
   ...
}
```
then this should compile without any changes. But if they did:
```
client_side::data_sources::DataSourceStatus status = sdk.DataSourceStatus().Status();
```
that's going to require a change.

We have [some](https://github.com/launchdarkly/cpp-sdks/releases/tag/launchdarkly-cpp-client-v3.0.6) [precedent ](https://github.com/launchdarkly/cpp-sdks/releases/tag/launchdarkly-cpp-client-v3.0.5)for "non-major to fix a bug in the API definition".

Reply:
Actually, this might be fine. Since `client_side::data_sources::DataSourceStatus` is a type alias, and the new type didn't exist, it should still compile. It's definitely ABI breaking but we don't claim to support a stable C++ ABI.

Reply:
The following code compiles without issue on the existing client, and the client in this PR:
```c++
  client_side::data_sources::DataSourceStatus ds_status =
        client.DataSourceStatus().Status();

    std::optional<client_side::data_sources::DataSourceStatus::ErrorInfo>
        last_err = ds_status.LastError();

    client_side::data_sources::DataSourceStatus::DataSourceState state =
        ds_status.State();

    client_side::data_sources::DataSourceStatus::DateTime date =
        ds_status.StateSince();
```